runtime/include/mem/cstdlib/chpl-mem-impl.h

```diff
@@ -37,7 +37,10 @@ static inline void* chpl_malloc(size_t size) {
 }
 
 static inline void* chpl_memalign(size_t boundary, size_t size) {
-  return sys_memalign(boundary, size);
+  void *rp = sys_memalign(boundary, size);
+  if(rp==NULL)
```

Comment:
NULL is a valid valid if `size == 0`, so I think we need to check `if (rp == NULL && size != 0)`

Might also want to use `ret` instead of `rp` to more closely match the style in other runtime memory functions